src/main/resources/swagger/api-docs.yaml

```diff
@@ -2123,6 +2123,39 @@ paths:
           description: Resource type does not exist or you are not a member of any
             policy on the resource
           content: { }
+  /api/resources/v2/{resourceTypeName}/{resourceId}/cascade:
+    delete:
+      tags:
+        - Resources
+      summary: Delete a resource
+      operationId: deleteResourceCascadeV2
+      parameters:
+        - name: resourceTypeName
+          in: path
+          description: Type of the resource
+          required: true
+          schema:
+            type: string
+        - name: resourceId
+          in: path
+          description: Id of the resource
+          required: true
+          schema:
+            type: string
+      responses:
+        204:
+          description: Successfully deleted resource
+          content: { }
+        400:
+          description: Cannot delete a resource with children. Delete the children first then try again.
+          content: { }
+        403:
+          description: You do not have permission to perform this action on the resource or permissions on the resource's parent
+          content: { }
+        404:
+          description: Resource type does not exist or you are not a member of any
+            policy on the resource
+          content: { }
```

Comment:
I know you expressed wanting to make this a separate API, but looking at it I am wondering again if it can be a flag/option on the existing delete-resource API

Reply:
It can, that would just cascade changes throughout the codebase.  I *think* if we make the flag optional it won't require updating everyone that calls the API but I'm not positive.